exchange/service.go

```diff
@@ -463,18 +463,11 @@ func processGetSelectedServicesResponse(mURL string, mOrg string, mVersion strin
 
 	// If the caller wanted a specific version, check for 1 result.
 	if searchVersion != "" {
-		if len(msMetadata) != 1 {
-			glog.Errorf(rpclogString(fmt.Sprintf("expecting 1 service %v %v %v response: %v", mURL, mOrg, mVersion, resp)))
-			return ret, errors.New(fmt.Sprintf("expecting 1 service %v %v %v, got %v", mURL, mOrg, mVersion, len(msMetadata)))
-		} else {
-			for msId, msDef := range msMetadata {
-				glog.V(3).Infof(rpclogString(fmt.Sprintf("returning service definition %v", msDef.ShortString())))
-				ret[msId] = msDef
-				return ret, nil
-			}
-			return ret, errors.New("should not get here")
+		for msId, msDef := range msMetadata {
```

Comment:
The comment above really needs to be updated. If I am reading this correctly, a search for a specific version can return multiple services because the arch field is empty, and thus multiple services can be returned.